streams/src/main/java/org/apache/kafka/streams/kstream/SessionWindows.java

```diff
@@ -187,7 +187,8 @@ public long inactivityGap() {
      */
     @Deprecated
     public long maintainMs() {
-        return Math.max(maintainDurationMs, gapMs);
+        final long gracePeriodMs = graceMs != -1 ? graceMs : Math.max(maintainDurationMs, gapMs) - inactivityGap();
```

Comment:
This is not fixing the issue.. we should use the `graceMs`  here, which may be specified by the user.

Reply:
@guozhangwang  Thank you for your guidance.
So I should change it to the following code. Is this correct?
```
    public long maintainMs() {
        return Math.max(maintainDurationMs, gapMs + graceMs);
    }
```

Reply:
Yes, that's right.